### PR TITLE
kmod: full cross-compatibility conan v1/v2 + add test_v1_package

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]

**Comment by SpaceIm** [comment]
It's worth noting that test_v1_package works because kmod is always shared, otherwise a static build would have fail: https://github.com/conan-io/conan-center-index/pull/12063#issuecomment-1206317836 (due to dependencies not propagated to cmake_find_package_multi when self.cpp_info.requires not explicitly populated with dependencies).

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/kmod/all/conanfile.py [hunk 1]
```diff
@@ -38,8 +37,14 @@ class KModConan(ConanFile):
     }
 
     def configure(self):
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        try:
+           del self.settings.compiler.libcxx
```
Comment (prince-chrismc) [inline recipes/kmod/all/conanfile.py:41]: Is there an issue for a "safe delete" feature so we dont need this mess?

Comment (SpaceIm) [reply recipes/kmod/all/conanfile.py:41]: https://github.com/conan-io/conan/issues/11792

Comment (uilianries) [reply recipes/kmod/all/conanfile.py:41]: Unfortunately not, I asked about it here: https://github.com/conan-io/conan/issues/11801

**Pull request merged**